tests/test_api.py

```diff
@@ -0,0 +1,103 @@
+import requests_mock
+from inquestlabs import inquestlabs_exception
+import pytest
+import sys
+import os
+import requests
+sys.path.insert(1, os.path.join(sys.path[0], '..'))
+
+
+def mocked_400_response_request(*args, **kwargs):
```

Comment:
Can we add a mock test for hitting the rate limit?

Reply:
Is it not the test on line 33?

Reply:
The test on line 33 is testing for num_retries with connections to the server. If the ratelimit hit is covered by an inquestlabs_exception, then it is already tested for.

Reply:
Rate limit has a special return code of 429. That's why I was saying we should mock it explicitly.